### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by Woody4618**
- Add priority fee parameter to upload so that it is more reliable 
- Fixed the abort signal so that its possible to stop the --remote command 
- Added as skip build parameter to verify-from-repo so that one does not have to wait until the build is done to upload the on chain data. For example in case the upload fails or there is some other error in between
- Check if program is actually deployed on the current cluster and show an error if it is not

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -188,7 +228,11 @@ async fn main() -> anyhow::Result<()> {
             .arg(Arg::with_name("cargo-args")
                 .multiple(true)
                 .last(true)
-                .help("Arguments to pass to the underlying `cargo build-bpf` command")))
+                .help("Arguments to pass to the underlying `cargo build-bpf` command"))
+            .arg(Arg::with_name("skip-build")
+                .long("skip-build")
+                .help("Skip building and verification, only upload the PDA")
+                .takes_value(false)))
```
Comment (ngundotra) [inline src/main.rs:235]: This is what `solana-verify remote submit-job` is for

File: src/main.rs [hunk 2]
```diff
@@ -49,24 +49,58 @@ pub fn get_network(network_str: &str) -> &str {
...
+                    .output()
+                    .is_err()
+                {
+                    println!("Failed to remove temporary directory");
+                } else {
+                    println!("Removed temporary directory {}", temp_dir);
+                }
+            }
+
+            std::process::exit(130);
```
Comment (ngundotra) [inline src/main.rs:100]: why use this specific exit code?

File: src/main.rs [hunk 3]
```diff
@@ -895,114 +950,15 @@ pub async fn verify_from_repo(
...
+    mut skip_build: bool,
     container_id_opt: &mut Option<String>,
     temp_dir_opt: &mut Option<String>,
+    check_signal: &dyn Fn(&mut Option<String>, &mut Option<String>),
 ) -> anyhow::Result<()> {
-    if remote {
-        let genesis_hash = get_genesis_hash(connection)?;
-        if genesis_hash != MAINNET_GENESIS_HASH {
-            return Err(anyhow!("Remote verification only works with mainnet. Please omit the --remote flag to verify locally."));
-        }
```
Comment (ngundotra) [inline src/main.rs:905]: Why is this entire block getting deleted?

**Review by ngundotra — CHANGES_REQUESTED** [review]
Good fixes overall- will help devs troubleshoot when they haven't deployed their own program, or can't land a tx on mainnet url. Added some questions & small nits.

File: src/main.rs (hunk 2 above)
Comment (Woody4618) [reply src/main.rs:100]: Its the exit code for ctrl+c https://tldp.org/LDP/abs/html/exitcodes.html

Comment (Woody4618) [reply src/main.rs:100]: Ill remove that an see of just breaking the loop is also enough. I had a hard time getting this endless API request to stop.

File: src/main.rs (hunk 3 above)
Comment (Woody4618) [reply src/main.rs:905]: It was duplicate code exactly the same as 100 lines below. Both code paths now use the same args logic

File: src/main.rs (hunk 1 above)
Comment (ngundotra) [reply src/main.rs:235]: Okay this makes sense.

verify-from-repo --remote will now trigger write to PDA & submit job, skipping build

File: src/main.rs (hunk 3 above)
Comment (ngundotra) [reply src/main.rs:905]: I think you forgot to add this get_genesis_hash check back in

**Review by ngundotra — CHANGES_REQUESTED** [review]
One last note, all other nits fixed 🫡

**Comment by ngundotra** [comment]
Looking at this again, I think this may also solve an error that occurs if you attempt to run `verify-from-repo --remote` before running `verify-from-repo` locally.

This error will be thrown when attempting to fill --library-name when none is provided. We attempt to run `find` in a repo, but that repo will not exist on user's file system until we clone the repo. Which does not happen on `--remote`.

Would appreciate if you could check that `--remote` works even if you haven't run `verify-from-repo` locally for that url

File: src/main.rs (hunk 3 above)
Comment (Woody4618) [reply src/main.rs:905]: Its still there, no? In line 1140

Comment (ngundotra) [reply src/main.rs:905]: You're right! Looks good to me

**Review by ngundotra — APPROVED** [review]
Ln 1140 has get_genesis_hash check

**Review by manslwk — COMMENTED** [review]
Kingo

**Review by manslwk — COMMENTED** [review]
Limón

**Review by manslwk — COMMENTED** [review]
Kingo

**Pull request merged**